handlers.go

```diff
@@ -81,3 +95,26 @@ func (dh *DeviceHandlers) writeObject(w http.ResponseWriter, device *Device) {
 		return
 	}
 }
+
+func pageAndLimitWrapper(h http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		limit, err := readIntFromQueryParameter(r.URL, "limit", 100)
+		if err != nil {
+			fmt.Println(err.Error())
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		page, err := readIntFromQueryParameter(r.URL, "page", 0)
+		if err != nil {
+			fmt.Println(err.Error())
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		for key, val := range map[string]int{"limit": limit, "page": page} {
+			r = r.WithContext(context.WithValue(r.Context(), key, val))
+		}
+
+		h.ServeHTTP(w, r)
```

Comment:
I think this would be more readable:

```
ctx := context.WithValue(r.Context(), "page", page)
ctx = context.WithValue(ctx, "limit", limit)
h.ServeHTTP(w, r.WithContext(ctx))
```